ckanext/validation/jobs.py

```diff
@@ -395,3 +410,68 @@ def _prep_foreign_keys(package, table_schema, resource, df):
         for field in table_schema['fields']:
             if field['name'] in foreign_keys.keys():
                 field['foreignKey'] = foreign_keys[field['name']]
+
+
+def _reorder_columns(schema, df):
+    """
+    Creating a preset in which you can configure whether the order of columns matters.
+    """
+
+    log.debug(df.dtypes)
+    log.debug("Considering whether to switch column order")
+
+    required_field_order = list(map(
+        lambda x: x['name'],
+        schema.get('fields', [])
+    ))
+    submitted_field_order = list(df.columns)
+    errors = {}
+    for f in required_field_order:
+        if f in submitted_field_order:
+            submitted_field_order.remove(f)
+        else:
+            df[f] = pandas.np.NaN
+            error_key = "Missing {} field".format(f)
+            error_message = "Uploaded data file is missing required field \"{}\"".format(f)
+            errors[error_key] = [error_message]
+
+    if errors:
+        raise t.ValidationError(errors)
+
+    new_column_order = list(required_field_order + submitted_field_order)
+    old_column_order = list(df.columns)
+    column_mapping = {}
+
+    log.debug("New Order: " + str(new_column_order))
+    log.debug("Old Order: " + str(old_column_order))
+
+    if not new_column_order == old_column_order:
+        log.debug("Switching column order")
+        df = df[new_column_order]
+        log.debug(
+            'The fields have had to be reordered to pass validation. '
+            'Column numbers in error messages may be wrong!'
+        )
+        for i, col in enumerate(old_column_order):
+            column_mapping[i+1] = new_column_order.index(col)+1
```

Comment:
Not sure about this one but spaces next to operators? `i + 1` instead of `i+1`

Reply:
I think pep8 says that inside brackets there should not be spaces.  But there should be spaces around the +1 at the end of the line.